src/core/Tree/TreeNode.tsx

```diff
@@ -0,0 +1,257 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { CommonProps, getWindow, useTheme } from '../utils';
+import '@itwin/itwinui-css/css/tree.css';
+import { SvgChevronRight } from '@itwin/itwinui-icons-react';
+import { IconButton } from '../Buttons/IconButton';
+import cx from 'classnames';
+import { TreeContext, TreeData } from './Tree';
+
+export type TreeNodeProps = {
+  /**
+   * Unique id of the node.
+   */
+  nodeId: string;
+  /**
+   * The main text displayed on the node.
+   */
+  label?: React.ReactNode;
+  /**
+   * Small note displayed below main label.
+   */
+  sublabel?: React.ReactNode;
+  /**
+   * Icon shown before label and sublabel content.
+   */
+  icon?: JSX.Element;
+  /**
+   * Is TreeNode disabled.
+   * @default false
+   */
+  isDisabled?: boolean;
+  /**
+   * Callback fired when expanding or closing a TreeNode.
+   */
+  onNodeExpanded?: (nodeId: string, expanded: boolean) => void;
+  /**
+   * Callback fired when selecting a TreeNode.
+   */
+  onNodeSelected?: (nodeId: string, selected: boolean) => void;
+  /**
+   * The TreeNode's child nodes.
+   * If undefined or empty, expander button is not shown.
+   */
+  subNodes?: Array<TreeData>;
+  /**
+   * Checkbox to be shown before TreeNode.
+   * If undefined checkbox will not be shown.
+   * Recommended to use Checkbox component.
+   */
+  nodeCheckbox?: React.ReactNode;
+  /**
+   * Callback fired when checking a TreeNode's checkbox.
+   */
+  onNodeCheckboxSelected?: (nodeId: string, checked: boolean) => void;
```

Comment:
This callback is not needed as users pass checkbox themselves therefore they can have `onChanged` props on it.

Reply:
Removed